conf/cassandra-env.sh

```diff
@@ -131,9 +131,11 @@ fi
 if [ "x$MAX_HEAP_SIZE" = "x" ] && [ "x$HEAP_NEWSIZE" = "x" -o $USING_G1 -eq 0 ]; then
     MAX_HEAP_SIZE="$CALCULATED_MAX_HEAP_SIZE"
     HEAP_NEWSIZE="$CALCULATED_CMS_HEAP_NEWSIZE"
-elif [ "x$MAX_HEAP_SIZE" = "x" ] ||  [ "x$HEAP_NEWSIZE" = "x" -a $USING_G1 -ne 0 ]; then
-    echo "please set or unset MAX_HEAP_SIZE and HEAP_NEWSIZE in pairs when using CMS GC (see cassandra-env.sh)"
-    exit 1
+elif [ $USING_CMS -eq 0 ]; then
```

Comment:
after changes in bin/nodetool where I parse it from this script, these changes in particular are not entirely necessary but I like to be explicit about CMS. So far we operate with logic "if it is not G1 then it is CMS". What if we go to use yet another GC? Then it would be triggered as well. If we explicitly say that "when using CMS GC" is the message before we exit, then we should explicitly check we are using CMS.

Reply:
e.g. what happens in this code when we use ZGC? So, obviously, USING_CMS branch will not be executed.

Then this 

````
if [ "x$MAX_HEAP_SIZE" = "x" ] && [ "x$HEAP_NEWSIZE" = "x" -o $USING_G1 -eq 0 ]; then
    MAX_HEAP_SIZE="$CALCULATED_MAX_HEAP_SIZE"
    HEAP_NEWSIZE="$CALCULATED_CMS_HEAP_NEWSIZE"
````

USING_G1 will be "1" (on ZGC) so the logic will be triggered only in case MAX_HEAP_SIZE is not set (which it is not by default) and when HEAP_NEWSIZE is not set either.

Reply:
if cms, then `MAX_HEAP_SIZE` and `HEAP_NEWSIZE` must both be set or both be not set.

if not cms (g1/zcs/shenendoah), then `HEAP_NEWSIZE` must never be set.  (we have to make sure that -XX:G1NewSizePercent=XX is always then set)

but: if nodetool isn't calling cassandra-env.sh anymore, then why patch cassandra-env.sh at all (especially in 5.0 branch)?

Reply:
I have remove this from what I just forcepushed.